### PR TITLE
Use gzip

### PR DESCRIPTION
**Pull request by VinayGuthal**

### CODE REVIEW

File: firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatController.java [hunk 1]
```diff
@@ -99,7 +101,11 @@
             JSONObject output = new JSONObject();
             output.put("heartbeats", array);
             output.put("version", "2");
-            return Base64.encodeToString(output.toString().getBytes(), Base64.DEFAULT);
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            GZIPOutputStream gzip = new GZIPOutputStream(out);
+            gzip.write(output.toString().getBytes());
+            gzip.close();
+            return Base64.encodeToString(out.toString("UTF-8").getBytes(), Base64.URL_SAFE);
```
Comment (carlonzo) [inline firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatController.java:108]: dont you need to close `out` as well?

Comment (VinayGuthal) [reply firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatController.java:108]: It is safe to not flush() or not close() a ByteArrayOutputStream. it makes no difference to memory usage

**Review by vkryachko — APPROVED** [review]

**Pull request merged**